app/components/file_attachment_list/file_attachment_list.js

```diff
@@ -124,61 +121,130 @@ export default class FileAttachmentList extends Component {
         previewImageAtIndex(this.items, idx, this.galleryFiles);
     });
 
-    renderItems = () => {
-        const {canDownloadFiles, fileIds, files} = this.props;
+    imagesSet = (files) => (files.filter((file) => file.has_preview_image));
 
-        if (!files.length && fileIds.length > 0) {
-            return fileIds.map((id, idx) => (
-                <FileAttachment
-                    key={id}
-                    canDownloadFiles={canDownloadFiles}
-                    file={{loading: true}}
-                    id={id}
-                    index={idx}
-                    theme={this.props.theme}
-                />
-            ));
+    nonImagesSet = (files) => (files.filter((file) => !file.has_preview_image));
+
+    imagesCount = () => {
+        const {files} = this.props;
+        return this.imagesSet(files).length;
+    }
+
+    moreImagesCount = () => {
+        const {files} = this.props;
+        const value = this.imagesSet(files).length - MAX_VISIBLE_ROW_IMAGES;
+
+        if (value > 0) {
+            return value;
         }
 
-        return files.map((file, idx) => {
+        return null;
+    }
+
+    attachmentIndex = (fileId) => {
+        const {files} = this.props;
+        return files.findIndex((file) => file.id === fileId);
+    }
+
+    renderItems = (items) => {
+        const {canDownloadFiles} = this.props;
+        const imagesCount = this.imagesCount();
+        const moreImagesCount = this.moreImagesCount();
+
+        return items.map((file, idx) => {
             const f = {
                 caption: file.name,
                 data: file,
             };
 
+            const imagesListProps = {imagesCount};
+            if ((idx === MAX_VISIBLE_ROW_IMAGES - 1) && moreImagesCount) {
+                imagesListProps.moreImagesCount = moreImagesCount;
+            }
+
             return (
-                <FileAttachment
+                <View
+                    style={styles.pictureFrame}
                     key={file.id}
-                    canDownloadFiles={canDownloadFiles}
-                    file={f}
-                    id={file.id}
-                    index={idx}
-                    onCaptureRef={this.handleCaptureRef}
-                    onPreviewPress={this.handlePreviewPress}
-                    onLongPress={this.props.onLongPress}
-                    theme={this.props.theme}
-                />
+                >
+                    <FileAttachment
+                        key={file.id}
+                        canDownloadFiles={canDownloadFiles}
+                        file={f}
+                        id={file.id}
+                        index={this.attachmentIndex(file.id)}
```

Comment:
`idx` is the array index for the row being rendered. i.e. for the first pass, `idx` will be the index for the collected array of images, and subsequently, `idx` will be the index for the collected array of non-images.

The trouble arises in the attachment viewer window for a scenario with mixed attachment types. For example: `[Image, nonImage, Image, Image]` in that order. Without looking up the "original" index in the master attachments list, the attachment viewer opens the wrong item when the non-image attachment item is touched (as shown below):

<img width="200" alt="wrong index" src="https://user-images.githubusercontent.com/887849/68497979-9041ae80-0234-11ea-98f0-8fddb85928df.gif">

Reply:
On this I would say even if it is not related to the ticket perhaps it makes sense to remove other file types from the viewer

Reply:
This will make life easier for sure. On it.

Reply:
@enahum So, I can't remove non-image attachments from the viewer completely, since that removes the ability to "Download" these attachments via the 3-button icon.

Technical debt for subsequent analysis? For what it's worth, I'm not happy either with this peeking-back at the master attachment list just to keep the viewer functionality happy. 😞